internal/test/scorecard/common_utils.go

```diff
@@ -339,6 +383,62 @@ func createAndWaitTillCryostatAvailable(cr *operatorv1beta1.Cryostat, resources
 	return cr, nil
 }
 
+func createAndWaitTillClusterCryostatAvailable(cr *operatorv1beta1.ClusterCryostat, resources *TestResources) (*operatorv1beta1.ClusterCryostat, error) {
+	client := resources.Client
+	r := resources.TestResult
+
+	cr, err := client.OperatorCRDs().ClusterCryostats(cr.Spec.InstallNamespace).CreateCluster(context.Background(), cr)
+	if err != nil {
+		logError(r, fmt.Sprintf("failed to create ClusterCryostat CR: %s", err.Error()))
+		return nil, err
+	}
+
+	// Poll the deployment until it becomes available or we timeout
+	ctx, cancel := context.WithTimeout(context.Background(), testTimeout)
+	defer cancel()
+	err = waitForDeploymentAvailability(ctx, client, cr.Spec.InstallNamespace, cr.Name, r)
+	if err != nil {
+		logError(r, fmt.Sprintf("ClusterCryostat main deployment did not become available: %s", err.Error()))
+		return nil, err
+	}
+
+	err = wait.PollImmediateUntilWithContext(ctx, time.Second, func(ctx context.Context) (done bool, err error) {
+		cr, err = client.OperatorCRDs().ClusterCryostats(cr.Spec.InstallNamespace).GetCluster(ctx, cr.Name)
+		if err != nil {
+			return false, fmt.Errorf("failed to get ClusterCryostat CR: %s", err.Error())
+		}
+		if len(cr.Status.TargetNamespaces) > 0 && cr.Status.TargetNamespaces[0] == cr.Spec.TargetNamespaces[0] {
+			r.Log += fmt.Sprintf("application has access to the following namespace: %s\n", cr.Status.TargetNamespaces)
+			return true, nil
+		}
+		r.Log += "application's target namespaces are not yet available\n"
+		return false, nil
+	})
+	if err != nil {
+		logError(r, fmt.Sprintf("application's target namespaces not found in CR: %s", err.Error()))
+		return nil, err
+	}
+
+	err = wait.PollImmediateUntilWithContext(ctx, time.Second, func(ctx context.Context) (done bool, err error) {
+		cr, err = client.OperatorCRDs().ClusterCryostats(cr.Spec.InstallNamespace).GetCluster(ctx, cr.Name)
+		if err != nil {
+			return false, fmt.Errorf("failed to get ClusterCryostat CR: %s", err.Error())
+		}
+		if len(cr.Status.CryostatStatus.ApplicationURL) > 0 {
+			return true, nil
+		}
+		r.Log += "application URL is not yet available\n"
+		return false, nil
+	})
+	if err != nil {
+		logError(r, fmt.Sprintf("application URL not found in CR: %s", err.Error()))
+		return nil, err
+	}
+	r.Log += fmt.Sprintf("application is available at %s\n", cr.Status.CryostatStatus.ApplicationURL)
```

Comment:
I think we can also check both the application URL and targetNamespaces status on the wait loop above this?